File: lib/layer2_core/light/e_tertiary_fixed_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ETertiaryFixedLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl37;
      case 'standard': return cstl37;
      case 'blue': return cbl37;
      case 'green': return cgl37;
      case 'red': return crl37;
      case 'nothing': return cnol37;
      default: return cstl37;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ETertiaryFixedLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ETertiaryFixedLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl37
            AppTheme.STANDARD -> R.color.cstl37
            AppTheme.BLUE -> R.color.cbl37
            AppTheme.GREEN -> R.color.cgl37
            AppTheme.RED -> R.color.crl37
            AppTheme.NOTHING -> R.color.cnol37
        }
    }
}
